src/packages/solid/components/Input/Input.tsx

```diff
@@ -15,90 +15,122 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
-import { type Component, type Signal } from 'solid-js';
+import { createSignal, type Component, type Signal, createEffect, on } from 'solid-js';
 import { View, Text, type IntrinsicNodeProps } from '@lightningjs/solid';
 import styles, { type InputStyles } from './Input.styles.js';
-import type { KeyHandler } from '@lightningjs/solid-primitives';
 import type { Tone } from '../../types/types.js';
 
 export interface InputProps extends IntrinsicNodeProps {
   /**
-   * actualTitle is a proxy element for title which is used to store actual value typed when password is enabled
+   * non-reactive index of the current cursor currentPosition
    */
-  actualTitle?: string;
-
-  /**
-   * eyebrow of input container.
-   */
-  eyebrow?: string;
+  position?: number;
 
   /**
-   * help text for the input container.
+   * signal passed in to represent what change we want to happe in the input
    */
-  helpText?: string;
+  keyEvent: Signal<string>;
 
   /**
-   * index of the current cursor positions
+   * signal passed in to represent the actual title within the input
    */
-  position?: number;
-
-  keySignal: Signal<string>;
-
-  onRight?: KeyHandler;
-
-  onLeft?: KeyHandler;
+  titleSignal: Signal<string>;
 
   style?: Partial<InputStyles>;
 
   tone?: Tone;
 }
 
 const Input: Component<InputProps> = props => {
+  const [title, setTitle] = props.titleSignal;
+  const [position, setPosition] = createSignal(props.position ?? title().length);
+  const [keyEvent, setkeyEvent] = props.keyEvent;
```

Comment:
can we add eslint-ignores to these since we're expecting them to be non-reactive

Reply:
the k in `setkeyEvent` should be capitalized